src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1108,7 +1108,7 @@ protected void GuardedAddBoundTreeForStandaloneSyntax(CSharpSyntaxNode syntax, B
 
             if (!alreadyInTree)
             {
-                if ((this.IsSpeculativeSemanticModel && syntax == _root) || syntax is StatementSyntax)
+                if ((this.IsSpeculativeSemanticModel && syntax == _root) || syntax is StatementSyntax || syntax is ArrowExpressionClauseSyntax)
```

Comment:
What is the ```_root``` in this scenario? Perhaps we can simply remove the ```this.IsSpeculativeSemanticModel``` condition instead of having a special condition for ArrowExpressionClauseSyntax. This might serve us better in the long run. Also, we prefer Kind checks to type checks, type check is used for StatementSyntax because StatementSyntax is a base class with a lot of derived classes.